content/reliability/docs/architecting-for-static-stability.md

```diff
@@ -0,0 +1,19 @@
+
+First is the assumption that capacity will be available in other Availability Zones during the outage. Correlated events can affect other customer workloads as well. If every workload evacuates simultaneously, it will cause a large spike in demand for capacity in the other Availability Zones. Most times, large capacity pools in EC2 will absorb the demand, but risks increase for higher demand instance types and smaller regions.
+
+Second is the assumption that the correlated outage will not impact control plane orchestration systems. While these systems are designed to be highly available, the number of systems involved in a failover scenario spans multiple AWS services, Kubernetes core components, and user installed Kubernetes operators. Evacuation results in a large spike in traffic across these systems, and correlated impact on any of these systems can cause an evacuation strategy to fail.
+
+Third is the introduction of bi-modal system behavior. Availability Zone evacuation is a rare event and introduces a new operational mode in addition to the standard operational mode of the system. This mechanism is expensive to test, and thus, rarely tested. Changes to the system over time may violate original assumptions and test results, the impact of which won’t surface until it’s too late.
+
+Static stability solves this problem and is a recommended best practice in both AWS and Kubernetes. Consider spreading application replicas across multiple failure domains, including across virtual machines and Availability Zones. In the event of a correlated outage, the automation halts changes, allowing the existing capacity to handle the workload’s traffic. Networking layers shift traffic away from the impacted Availability Zone towards the replicas in healthy Availability Zones. Application scaling should correspond to its reliability requirements. To survive a single Availability Zone outage, replicas in other AZs must be preemptively scaled to handle the additional load. Workload performance can be nonlinear, so it’s essential to test. To survive a multiple availability zone event, larger overhead is required. At a fundamental level, customers must choose the right balance of redundancy and cost that meets their use case.
```

Comment:
This part feels a little hand wavy to me. I get that you should use topology spread and anti-affinity to distribute pods across your infrastructure, but how do you shift traffic away for the impacted AZ? How do you preemptively scale replicas in the non-affected AZs? Will the scheduler (and Karpenter) know to avoid provisioning capacity in the affected AZ?

Reply:
> Will the scheduler (and Karpenter) know to avoid provisioning capacity in the affected AZ?

Not currently. There's probably a followon for AZ resiliency as we mature this across systems.